===================================================================
RCS file: /cvsroot/kino/dvgrab/avi.h,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- kino/dvgrab/avi.h	2005/04/18 16:36:07	1.7
+++ kino/dvgrab/avi.h	2006/03/04 05:29:38	1.8
@@ -35,8 +35,8 @@
 
 #define AVI_SMALL_INDEX (0x01)
 #define AVI_LARGE_INDEX (0x02)
-#define AVI_INDEX_OF_INDEXES (0x00)
-#define AVI_INDEX_OF_CHUNKS (0x01)
+#define KINO_AVI_INDEX_OF_INDEXES (0x00)
+#define KINO_AVI_INDEX_OF_CHUNKS (0x01)
 #define AVI_INDEX_2FIELD (0x01)
 
 enum { AVI_PAL, AVI_NTSC, AVI_AUDIO_48KHZ, AVI_AUDIO_44KHZ, AVI_AUDIO_32KHZ };

===================================================================
RCS file: /cvsroot/kino/dvgrab/avi.cc,v
retrieving revision 1.11
retrieving revision 1.12
diff -u -r1.11 -r1.12
--- kino/dvgrab/avi.cc	2005/04/18 16:36:07	1.11
+++ kino/dvgrab/avi.cc	2006/03/04 05:29:38	1.12
@@ -267,7 +267,7 @@
 	{
 		indx[ i ] ->wLongsPerEntry = 4;
 		indx[ i ] ->bIndexSubType = 0;
-		indx[ i ] ->bIndexType = AVI_INDEX_OF_INDEXES;
+		indx[ i ] ->bIndexType = KINO_AVI_INDEX_OF_INDEXES;
 		indx[ i ] ->nEntriesInUse = 0;
 		indx[ i ] ->dwReserved[ 0 ] = 0;
 		indx[ i ] ->dwReserved[ 1 ] = 0;
@@ -779,7 +779,7 @@
 
 	ix[ stream ] ->wLongsPerEntry = 2;
 	ix[ stream ] ->bIndexSubType = 0;
-	ix[ stream ] ->bIndexType = AVI_INDEX_OF_CHUNKS;
+	ix[ stream ] ->bIndexType = KINO_AVI_INDEX_OF_CHUNKS;
 	ix[ stream ] ->nEntriesInUse = 0;
 	ix[ stream ] ->dwChunkId = indx[ stream ] ->dwChunkId;
 	ix[ stream ] ->qwBaseOffset = offset + length;
@@ -1038,7 +1038,7 @@
 	   several RIFF chunks of less than 1 GByte size. Old software
 	   however would only deal with the first RIFF chunk.
 
-	   Note that the first entry (FILE) isn´t actually part
+	   Note that the first entry (FILE) isnt actually part
 	   of the AVI file. I use this (pseudo-) directory entry to
 	   keep track of the RIFF chunks and their positions in the
 	   AVI file.

===================================================================
RCS file: /cvsroot/kino/dvgrab/config.h.in,v
retrieving revision 1.5
retrieving revision 1.6
diff -u -r1.5 -r1.6
--- kino/dvgrab/config.h.in	2005/04/18 16:36:07	1.5
+++ kino/dvgrab/config.h.in	2006/03/04 05:29:38	1.6
@@ -9,9 +9,6 @@
 /* Define to 1 if you have the <jpeglib.h> header file. */
 #undef HAVE_JPEGLIB_H
 
-/* Define to 1 if you have the `a52' library (-la52). */
-#undef HAVE_LIBA52
-
 /* Define to 1 if you have the `avc1394' library (-lavc1394). */
 #undef HAVE_LIBAVC1394
 
@@ -24,39 +21,21 @@
 /* Define to 1 if you have the `jpeg' library (-ljpeg). */
 #undef HAVE_LIBJPEG
 
-/* Define to 1 if you have the `mpeg3' library (-lmpeg3). */
-#undef HAVE_LIBMPEG3
-
-/* Define to 1 if you have the `ogg' library (-logg). */
-#undef HAVE_LIBOGG
-
-/* Define to 1 if you have the `png' library (-lpng). */
-#undef HAVE_LIBPNG
-
 /* Define to 1 if you have the `pthread' library (-lpthread). */
 #undef HAVE_LIBPTHREAD
 
-/* Define to 1 if you have the `quicktime' library (-lquicktime). */
+/* libquicktime.sourceforge.net present */
 #undef HAVE_LIBQUICKTIME
 
 /* Define to 1 if you have the `rom1394' library (-lrom1394). */
 #undef HAVE_LIBROM1394
 
-/* Define to 1 if you have the `vorbis' library (-lvorbis). */
-#undef HAVE_LIBVORBIS
-
-/* Define to 1 if you have the `vorbisenc' library (-lvorbisenc). */
-#undef HAVE_LIBVORBISENC
-
 /* Define to 1 if you have the <memory.h> header file. */
 #undef HAVE_MEMORY_H
 
 /* Define to 1 if you have the `mktime' function. */
 #undef HAVE_MKTIME
 
-/* Define to 1 if you have the <quicktime/lqt.h> header file. */
-#undef HAVE_QUICKTIME_LQT_H
-
 /* Define to 1 if you have the <quicktime/quicktime.h> header file. */
 #undef HAVE_QUICKTIME_QUICKTIME_H
 

===================================================================
RCS file: /cvsroot/kino/dvgrab/configure.in,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -r1.15 -r1.16
--- kino/dvgrab/configure.in	2005/09/25 08:43:42	1.15
+++ kino/dvgrab/configure.in	2006/03/04 05:29:38	1.16
@@ -41,28 +41,36 @@
 	AC_ERROR(You need the pthread library to compile dvgrab)
 ])
 
-AC_CHECK_HEADERS(quicktime/quicktime.h,,
+# LIBQUICKTIME
+PKG_CHECK_MODULES(LIBQUICKTIME, [libquicktime >= 0.9.5],
 [
-	AC_WARN(quicktime headers missing; install libquicktime or quicktime4linux to support Quicktime files.)
-])
-
-# add various libs if using quicktime4linux and not libquicktime
-AC_CHECK_HEADERS(quicktime/lqt.h,
-	[AC_CHECK_LIB(quicktime, quicktime_open, ,
-		[AC_WARN(libquicktime missing; install libquicktime or quicktime4linux to support Quicktime files.)]
-	)],
-[
-	AC_CHECK_LIB(png, png_read_image)
-	AC_CHECK_LIB(ogg, ogg_stream_pagein)
-	AC_CHECK_LIB(vorbis, vorbis_info_clear)
-	AC_CHECK_LIB(vorbisenc, vorbis_encode_ctl)
-	AC_CHECK_LIB(jpeg, jpeg_read_scanlines)
-	AC_CHECK_LIB(a52, a52_init)
-	AC_CHECK_LIB(mpeg3, mpeg3demux_read_char_packet)
-	AC_CHECK_LIB(quicktime, quicktime_open, ,
-		[AC_WARN(libquicktime missing; install libquicktime or quicktime4linux to support Quicktime files.)]
-	)
+	AC_DEFINE(HAVE_LIBQUICKTIME, 1, [libquicktime.sourceforge.net present])
+	AC_SUBST(LIBQUICKTIME_CFLAGS)
+	AC_SUBST(LIBQUICKTIME_LIBS)
+],[
+	AC_CHECK_HEADERS(quicktime/quicktime.h,
+	[
+		LIBQUICKTIME_CFLAGS="-I$prefix/include/quicktime"
+		AC_CHECK_LIB(quicktime, quicktime_open,
+		[
+			LIBQUICKTIME_LIBS="-lquicktime"
+			AC_DEFINE(HAVE_LIBQUICKTIME, 1, [Heroine Virtual Quicktime4Linux present])
+		],[
+			AC_CHECK_LIB(quicktimehv, quicktime_open,
+			[
+				LIBQUICKTIME_LIBS="-lquicktimehv"
+				AC_DEFINE(HAVE_LIBQUICKTIME, 1, [cvs.cinelerra.org Quicktime4Linux present])
+			],[
+				AC_WARN([libquicktime missing; install libquicktime or quicktime4linux to support Quicktime files.])
+			])
+		])
+	],[
+		AC_WARN($LIBQUICKTIME_PKG_ERRORS)
+		AC_WARN([libquicktime missing; install libquicktime or quicktime4linux to support Quicktime files.])
+	])
 ])
+AC_SUBST(LIBQUICKTIME_CFLAGS)
+AC_SUBST(LIBQUICKTIME_LIBS)
 
 AC_CHECK_HEADERS(jpeglib.h,,
 [

===================================================================
RCS file: /cvsroot/kino/dvgrab/filehandler.cc,v
retrieving revision 1.13
retrieving revision 1.14
diff -u -r1.13 -r1.14
--- kino/dvgrab/filehandler.cc	2005/04/18 16:36:07	1.13
+++ kino/dvgrab/filehandler.cc	2006/03/04 05:29:38	1.14
@@ -788,10 +788,9 @@
 bool QtHandler::Create( const string& filename )
 {
 	Init();
-
 	fd = quicktime_open( const_cast<char*>( filename.c_str() ), 0, 1 );
+	this->filename = filename;
 
-	//if (fd != NULL) FileTracker::GetInstance().Add(filename.c_str());
 	return ( fd != NULL );
 }
 
@@ -907,7 +906,7 @@
 	if ( fd )
 	{
 		struct stat file_status;
-		fstat( fileno( fd->stream ), &file_status );
+		stat( filename.c_str(), &file_status );
 		return file_status.st_size;
 	}
 	else
@@ -945,6 +944,7 @@
 		Close();
 		return false;
 	}
+	filename = s;
 
 	return true;
 }

===================================================================
RCS file: /cvsroot/kino/dvgrab/filehandler.h,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- kino/dvgrab/filehandler.h	2005/04/18 16:36:07	1.7
+++ kino/dvgrab/filehandler.h	2006/03/04 05:29:38	1.8
@@ -195,7 +195,7 @@
 };
 
 #ifdef HAVE_LIBQUICKTIME
-#include <quicktime/quicktime.h>
+#include <quicktime.h>
 
 class QtHandler: public FileHandler
 {

===================================================================
RCS file: /cvsroot/kino/dvgrab/Makefile.am,v
retrieving revision 1.10
retrieving revision 1.11
diff -u -r1.10 -r1.11
--- kino/dvgrab/Makefile.am	2005/04/18 16:36:07	1.10
+++ kino/dvgrab/Makefile.am	2006/03/04 05:29:38	1.11
@@ -15,12 +15,14 @@
 INCLUDES =	\
 	@LIBRAW1394_CFLAGS@ \
 	@LIBIEC61883_CFLAGS@ \
-	@LIBDV_CFLAGS@
+	@LIBDV_CFLAGS@ \
+	@LIBQUICKTIME_CFLAGS@
 
-dvgrab_LDFLAGS = \
+dvgrab_LDADD = \
 	@LIBRAW1394_LIBS@ \
 	@LIBIEC61883_LIBS@ \
-	@LIBDV_LIBS@
+	@LIBDV_LIBS@ \
+	@LIBQUICKTIME_LIBS@
 
 #riffdump_SOURCES = error.cc error.h riffdump.cc avi.h riff.h avi.cc riff.cc frame.h frame.cc
 

